packages/@lwc/synthetic-shadow/src/shared/node-collection-util.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { isUndefined, isTrue } from './language';
+/**
+ * Writing our own utils to handle NodeList and HTMLCollection. This is to not conflict with
+ * some legacy third party libraries like prototype.js that patch Array.prototype.
+ */
+
+/**
+ * Custom implementation of filter since using Array.prototype.filter conflicts with other
+ * legacy libraries like prototype.js
+ * Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter#Polyfill
+ */
+export function collectionFilter<T extends Node, K extends Element>(
+    this: NodeListOf<T> | HTMLCollectionOf<K>,
```

Comment:
if this are exported utility methods, they should not use `this`, instead they should receive the nodelist as the first argument. this applies to all these methods.